Library/Homebrew/dev-cmd/test.rb

```diff
@@ -93,9 +93,14 @@ def test
             exec(*args)
           end
         end
-      rescue ::Test::Unit::AssertionFailedError => e
+      rescue ChildProcessError => e
         ofail "#{f.full_name}: failed"
-        puts e.message
+        case e.inner["json_class"]
+        when "Test::Unit::AssertionFailedError"
+          puts e.inner["m"]
+        else
+          puts e.inner["json_class"], e.backtrace
```

Comment:
What happens to the exception message?

Reply:
`e.inner["m"]` contains the message, so it still gets printed in the `Test::Unit:AssertionFailedError` case.

Reply:
I duplicated the logic from below (the `rescue Exception => e`), since it's basically the same failure mode (non-assertion failures, it's just that we're distinguishing between non-assertion failures in the child and parent now).

I can make it print the message explicitly if you'd like, in which case I think it makes sense to update the original fallthrough too.

Reply:
Like above I think it might be nice to fiddle with handling `ChildProcessError` elsewhere so this special handling isn't required at all here.